permits/forms.py

```diff
@@ -257,3 +257,101 @@ def save(self, permit_request, commit=True):
             })
 
         return actor
+
+
+class PermitRequestActorForm(forms.ModelForm):
+
+
+    def __init__(self, *args, **kwargs):
+
+        request = kwargs.pop('request', None)
+        initial_values = kwargs.pop('initial', None)
+        print("**********************")
+        print(self)
+        permitrequestactor = models.PermitRequestActor.objects.filter(permit_request=initial_values['permit_request']).first()
+
+        if permitrequestactor:
+            initial = {**kwargs.get('initial', {}),
+            'actor_type': permitrequestactor.actor_type,
+            'description': permitrequestactor.description,
+            'name': permitrequestactor.actor.name,
+            'firstname': permitrequestactor.actor.firstname,
+            'company_name': permitrequestactor.actor.company_name,
+            'vat_number': permitrequestactor.actor.vat_number,
+            'address': permitrequestactor.actor.address,
+            'name': permitrequestactor.actor.name,
+            'city': permitrequestactor.actor.city,
+            'name': permitrequestactor.actor.name,
+            'phone_fixed': permitrequestactor.actor.phone_fixed,
+            'phone_mobile': permitrequestactor.actor.phone_mobile,
+            'name': permitrequestactor.actor.name,
+            }
+        else:
+
+            if initial_values:
+                initial = {'actor_type': initial_values['actor_type']}
+            else:
+                initial = {}
+
+        kwargs['initial'] = initial
+
+        super().__init__(*args, **kwargs)
+
+
+    name = forms.CharField(max_length=100)
+    firstname = forms.CharField( max_length=100)
+    company_name = forms.CharField(max_length=100)
+    vat_number = forms.CharField(max_length=100)
+    address = forms.CharField(max_length=100)
+    # zipcode = forms.PositiveIntegerField()
+    city = forms.CharField( max_length=100)
+    phone_fixed = forms.CharField(max_length=20)
+    phone_mobile = forms.CharField(max_length=20)
+    # email = forms.EmailField(_("email"), blank=True)
+
+
+    class Meta:
+        model = models.PermitRequestActor
+        exclude = ['actor', 'permit_request']
+        fields = ['actor_type','description', 'actor', ]
+
+    @transaction.atomic
+    def save(self, permit_request, commit=True):
+        print('***SAVING***')
+        name = self.cleaned_data.get('name')
```

Comment:
Tu pourrais ajouter un attribut sur le formulaire qui listerait les champs relatifs aux acteurs genre `user_fields = ['name', 'firstname', 'company_name', ...]`, comme ça tu t'épargnes pas mal de code: `actor, created = models.PermitActor.objects.update_or_create(**{field: self.cleaned_data.get(field) for field in self.user_fields})`.